app/Models/Site.php

```diff
@@ -23,20 +23,30 @@
 use Adshares\Adserver\Models\Traits\Ownership;
 use Illuminate\Database\Eloquent\Model;
 use Illuminate\Database\Eloquent\SoftDeletes;
+use Illuminate\Support\Collection;
 
 /**
  * @property int user_id
  * @property string name
  * @property array|null|string site_requires
  * @property array|null|string site_excludes
- * @property Zone[] zones
+ * @property Zone[]|Collection zones
  * @method static Site create($input = null)
  * @method static get()
  */
 class Site extends Model
 {
     use Ownership;
     use SoftDeletes;
+    public const STATUS_DRAFT = 0;
+    public const STATUS_INACTIVE = 1;
+    public const STATUS_ACTIVE = 2;
+    public const STATUSES = [self::STATUS_DRAFT, self::STATUS_INACTIVE, self::STATUS_ACTIVE];
+    private const ZONE_STATUS = [
+        Site::STATUS_DRAFT => Zone::STATUS_DRAFT,
+        Site::STATUS_INACTIVE => Zone::STATUS_ARCHIVED,
+        Site::STATUS_ACTIVE => Zone::STATUS_ACTIVE,
+    ];
```

Comment:
Why do we have different status for `Site` and `Zone` (`Inactive` vs `Archived`)? Imho it would be better to base on the same list of statuses. When a value is `1` it means `inactive` and that's all. 
In the future, we can move it to the external class/trait and use in banner and campaign model too.